test/error-messages/constraint-arg.txt

```diff
@@ -0,0 +1,12 @@
+Invalid value
+
+concat :: Array a -> Array a -> Array a
+          ^^^^^^^
+             1
+
+At position 1:
+  Where: 1st Argument
+  Value: /x/
+  Types: RegExp
+
+Expected types: (Array a).
```

Comment:
The expected type, in this case at least, reiterates the underlined portion of the type signature. Will this always be the case? If not, could you provide an example?

I favour something more minimal:

    Invalid value

    concat :: Array a -> Array a -> Array a
              ^^^^^^^
                 1

    1)  /x/ :: RegExp

    The value at position 1 is not a member of (Array a).